src/mainloop.cpp

```diff
@@ -186,7 +186,13 @@ void Mainloop::route_msg(struct buffer *buf)
             unknown = false;
             break;
         case Endpoint::AcceptState::Rejected:
-            // fall through
+            log_debug("Endpoint [%d] rejected message %u to %d/%d from %u/%u",
+                      e->fd,
+                      buf->curr.msg_id,
+                      buf->curr.target_sysid,
+                      buf->curr.target_compid,
+                      buf->curr.src_sysid,
+                      buf->curr.src_compid);
```

Comment:
Please add a `break;` statement after this log line as this is a full case and not just an intentional fall-through now.